cmd/promlts/query.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"net"
+	"net/http"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/improbable-eng/promlts/pkg/query"
+	"github.com/oklog/oklog/pkg/group"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/route"
+	"github.com/prometheus/prometheus/promql"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+// registerQuery registers a query command.
+func registerQuery(app *kingpin.Application, name string) runFunc {
+	cmd := app.Command(name, "query node exposing PromQL enabled Query API with data retrieved from multiple store nodes")
+
+	apiAddr := cmd.Flag("api-address", "listen address for the query API").
+		Default(":19099").String()
+
+	storeAddresses := cmd.Flag("store.addresses", "comma delimited listen addresses of store APIs").
+		Default("localhost:19090").Strings()
```

Comment:
I was wondering how we should go about discovery. Probably we'll want to support a plain list, DNS A + port, and DNS SRV. But we can add those capabilities later.